### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by ollavka**

### CODE REVIEW

File: src/averageAges.js [hunk 1]
```diff
@@ -14,12 +14,20 @@
...
+    return century
+      ? person.sex === 'm' && Math.ceil(person.died / 100) === century
+      : person.sex === 'm';
+  });
+
+  const avgAge = menList.reduce((sumAges, person) => {
+    const { born: bornYear, died: diedYear } = person;
+
+    return sumAges + diedYear - bornYear;
+  }, 0) / menList.length;
```
Comment (tetiana-ivanchenko) [inline src/averageAges.js:28]: move it to separate function and use it here and in the next function

File: src/averageAges.js [hunk 2]
```diff
@@ -36,8 +44,20 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
-function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+function calculateWomenAverageAge(people, withChildren = false) {
+  const womenList = people.filter(person => {
+    return withChildren
+      ? person.sex === 'f' && people.some(child => child.mother === person.name)
```
Comment (tetiana-ivanchenko) [inline src/averageAges.js:50]: if person id somebody's mother - no need to check her sex

File: src/averageAges.js [hunk 3]
```diff
@@ -54,8 +74,41 @@ function calculateWomenAverageAge(people, withChildren) {
...
+  }, []);
+
+  const avgAgeDiff = childrenWithMothers.reduce((sumAges, child) => {
+    const { mother, born: childBornYear } = child;
+    const { born: motherBornYear } = mother;
+
+    const ages = childBornYear - motherBornYear;
+
+    return sumAges + ages;
+  }, 0) / childrenWithMothers.length;
```
Comment (tetiana-ivanchenko) [inline src/averageAges.js:109]: combine this two reduce

**Review by tetiana-ivanchenko — CHANGES_REQUESTED** [review]

**Review by YegorKorenkov — APPROVED** [review]
Looks good

**Pull request closed without merging**